File: src/main/java/me/deecaad/weaponmechanicsplus/weapon/modifiers/attachments/Unlockable.java
package me.deecaad.weaponmechanicsplus.weapon.modifiers.attachments;

import me.deecaad.core.file.SerializeData;
import me.deecaad.core.file.Serializer;
import me.deecaad.core.file.SerializerException;
import me.deecaad.weaponmechanics.weapon.stats.WeaponStat;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

public class Unlockable implements Serializer<Unlockable> {

    private ItemStack itemStack;
    private int experience;
    private Map<WeaponStat, Number> stats;

    @NotNull
    @Override
    public Unlockable serialize(SerializeData data) throws SerializerException {
        return null;
    }
}

File: src/main/kotlin/com/cjcrafter/weaponmechanicsplus/weapon/modifiers/attachments/Unlockable.kt
package com.cjcrafter.weaponmechanicsplus.weapon.modifiers.attachments

import me.deecaad.core.file.*
import me.deecaad.weaponmechanics.weapon.stats.WeaponStat
import org.bukkit.inventory.ItemStack

class Unlockable : Serializer<Unlockable> {

    private val itemStack: ItemStack? = null
    private val experience = 0
    private val stats: Map<WeaponStat, Number>? = null

    @Throws(SerializerException::class)
    override fun serialize(data: SerializeData): Unlockable {
        return Unlockable()
    }
}